apps/glusterfs/app_node.go

```diff
@@ -320,6 +321,49 @@ func (a *App) NodeDelete(w http.ResponseWriter, r *http.Request) {
 				return err
 			}
 
+			volumes, err := ListCompleteVolumes(tx)
```

Comment:
I think this should only iterate over the volumes in the cluster that was changed rather than all volumes.